src/counterfactuals/core_struct.jl

```diff
@@ -30,7 +30,7 @@ end
 		initialization::Symbol = :add_perturbation,
 		generative_model_params::NamedTuple = (;),
 		min_success_rate::AbstractFloat=0.99,
-        converge_when::Symbol=:decision_threshold,
+        convergence::Union{AbstractConvergence,Symbol}=:decision_threshold,
         invalidation_rate::AbstractFloat=0.5,
```

Comment:
Redundancies such as the unneccessary keyword arguments here will be removed as part of another PR.